### PR TITLE
GITHUB-323: Deprecated Notebook's SQL table reasoning column no longer return Errors

### PR DESCRIPTION
**Pull request by manfromjupyter**
Switched Deprecated Notebook's SQL table reasoning column from t.string to t.text so will no longer return rails/MYSQL error: 'ActiveRecord::StatementInvalid - Data too long for column'

Test by either inspecting deprecate_notebooks table or adding a huge amount of text to a deprecation status.

Closes: #323

### CODE REVIEW

**Comment by mcrutch** [comment]
Should this also come with some kind of reasonable limit for the length?  That field in the database can now store 2GB of plain text and there isn't any limiting mechanism for it.
image

File: db/migrate/20201014151211_change_reasoning_to_be_text_in_deprecated_notebooks.rb
Comment (mcrutch) [inline db/migrate/20201014151211_change_reasoning_to_be_text_in_deprecated_notebooks.rb:3]: Went back and looked at the ticket.  Ticket talks about restricting length in Ruby.  This can't be merged without the corresponding piece otherwise the codebase is in a state where extremely large values can make it into the database (and once this is merged, it becomes part of the docker image).

**Review by mcrutch — CHANGES_REQUESTED** [review]

File: db/migrate/20201014151211_change_reasoning_to_be_text_in_deprecated_notebooks.rb
Comment (manfromjupyter) [reply db/migrate/20201014151211_change_reasoning_to_be_text_in_deprecated_notebooks.rb:3]: Oh shoot, I totally forgot about that. I agree, you're right, we definitely need a limit and error handling that enforces it. I'm on it.

**Comment by manfromjupyter** [comment]
@mcrutch, it should be good now. Added front end max character count of 500 characters that can entered in the input field and included a character count that appears below the form when you have 50 or less characters to use. At 500/500 characters it goes bold and red. And to prevent people trying to break it, did error handling from server side as well. Returns just a flash message with said error that it was too long. Was going to do the error inline and not refresh the page, but since this can only happen if they're trying to break it, hack it, etc, I thought this was the ideal solution.

**Comment by mcrutch** [comment]
Merge conflict, both changes are yours, so I will let you resolve the conflict.  Otherwise happy to merge once that is complete.

**Pull request merged**
